src/refresh-indicator.jsx

```diff
@@ -0,0 +1,263 @@
+    return (
+      <Paper
+        circle={true}
+        style={rootStyle}
+        ref="indicatorCt"
+      >
+        {this._renderChildren()}
+      </Paper>
+    );
+  },
+
+  _renderChildren() {
+    const paperSize = this._getPaperSize();
+    let childrenCmp = null;
+    if (this.props.status !== 'ready') {
+      const circleStyle = this._getCircleStyle(paperSize);
+      childrenCmp = (
+        <div ref="wrapper" style={this.mergeAndPrefix({
+            transition: Transitions.create('transform', '20s', null, 'linear'),
+            width: '100%',
+            height: '100%',
+          })}
+        >
+          <svg style={{
+              width: paperSize,
+              height: paperSize,
+            }}
+            viewBox={`0 0 ${VIEWBOX_SIZE} ${VIEWBOX_SIZE}`}
+          >
+            <circle ref="path"
+              style={this.mergeAndPrefix(circleStyle.style, {
+                transition: Transitions.create('all', '1.5s', null, 'ease-in-out'),
+              })}
+              {...circleStyle.attr}
+            />
+          </svg>
+        </div>
+      );
+    } else {
+      const circleStyle = this._getCircleStyle(paperSize);
+      const polygonStyle = this._getPolygonStyle(paperSize);
+      childrenCmp = (
+        <svg style={{
+            width: paperSize,
+            height: paperSize,
+          }}
+          viewBox={`0 0 ${VIEWBOX_SIZE} ${VIEWBOX_SIZE}`}
+        >
+          <circle
+            style={circleStyle.style}
+            {...circleStyle.attr}
+          >
+          </circle>
+          <polygon
+            style={polygonStyle.style}
+            {...polygonStyle.attr}
+          />
+        </svg>
+      );
+    }
+
+    return childrenCmp;
+  },
+
+  _getTheme() {
+    return this.context.muiTheme.component.refreshIndicator;
+  },
+
+  _getPaddingSize() {
+    const padding = this.props.size * 0.1;
+    return padding;
+  },
+
+  _getPaperSize() {
+    return this.props.size - this._getPaddingSize() * 2;
+  },
+
+  _getCircleAttr() {
+    return {
+      radiu: VIEWBOX_SIZE / 2 - 5,
+      originX: VIEWBOX_SIZE / 2,
+      originY: VIEWBOX_SIZE / 2,
+      strokeWidth: 3,
+    };
+  },
+
+  _getArcDeg() {
+    const p = this.props.percentage / 100;
+
+    const beginDeg = p * 120;
+    const endDeg = p * 410;
+    return [beginDeg, endDeg];
+  },
+
+  _getFactor() {
+    const p = this.props.percentage / 100;
+    const p1 = Math.min(1, p / 0.4);
+
+    return p1;
+  },
+
+  _getRootStyle() {
+    const padding = this._getPaddingSize();
+    return {
+      position: "absolute",
+      zIndex: 2,
+      width: this.props.size,
+      height: this.props.size,
+      padding: padding,
+      top: -10000,
+      left: -10000,
+      transform: `translate3d(${10000 + this.props.left}px, ${10000 + this.props.top}px, 0)`,
+      opacity: this.props.status === 'hide' ? 0 : 1,
+      transition: this.props.status === 'hide' ? Transitions.create('all', '.3s', 'ease-out') : 'none',
+    };
```

Comment:
Usually we would merge down `this.props.style` into the root styles. This would allow the user to override these variables. For example, overriding zIndex.